unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+{
+    struct ivi_layout_surface l_layoutSurface;
+    l_layoutSurface.id_surface = 1;
+    mp_iviShell->bkgnd_surface_id = 1;
+
+    struct weston_surface *lp_westonSurface = (struct weston_surface *)malloc(sizeof(struct weston_surface));
+    SET_RETURN_SEQ(surface_get_weston_surface, &lp_westonSurface, 1);
+
+    surface_event_create(&mp_iviShell->surface_created, &l_layoutSurface);
+
+    ASSERT_EQ(get_id_of_surface_fake.call_count, 1);
+    ASSERT_EQ(get_properties_of_surface_fake.call_count, 1);
+    ASSERT_EQ(wl_list_init_fake.call_count, 1);
+    ASSERT_EQ(surface_get_weston_surface_fake.call_count, 1);
+    ASSERT_EQ(wl_list_insert_fake.call_count, 1);
+    ASSERT_EQ(surface_add_listener_fake.call_count, 0);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 0);
+    ASSERT_EQ(0, g_SurfaceCreatedCount);
+    struct ivisurface *lp_iviSurf = (struct ivisurface*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct ivisurface, notification_list));
+    ASSERT_EQ(lp_iviSurf->shell, mp_iviShell);
+    ASSERT_EQ(lp_iviSurf->layout_surface, &l_layoutSurface);
+    ASSERT_EQ(mp_iviShell->bkgnd_surface, lp_iviSurf);
+
+    free(lp_iviSurf);
+    free(lp_westonSurface);
+}
+
+/** ================================================================================================
+ * @test_id             layer_event_create
+ * @brief               Test case of layer_event_create() where valid input params
+ * @test_procedure Steps:
+ *                      -# Set id layer of layout layer
+ *                      -# Calling the layer_event_create()
+ *                      -# Verification point:
+ *                         +# get_id_of_layer() must be called once time
+ *                         +# wl_list_insert() must be called once time
+ *                         +# wl_list_init() must be called once time
+ *                         +# get_properties_of_layer() must be called once time
+ *                         +# layer_add_listener() must be called once time
+ *                         +# wl_resource_post_event() must be called {MAX_NUMBER} times
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, layer_event_create)
+{
+    struct ivi_layout_layer l_layoutLayer;
+    l_layoutLayer.id_layer = 10;
+
+    layer_event_create(&mp_iviShell->layer_created, &l_layoutLayer);
+
+    ASSERT_EQ(get_id_of_layer_fake.call_count, 1);
+    ASSERT_EQ(wl_list_insert_fake.call_count, 1);
+    ASSERT_EQ(wl_list_init_fake.call_count, 1);
+    ASSERT_EQ(get_properties_of_layer_fake.call_count, 1);
+    ASSERT_EQ(layer_add_listener_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, MAX_NUMBER);
+    struct ivilayer *lp_iviLayer = (struct ivilayer*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct ivilayer, notification_list));
```

Comment:
get lp_iviLayer via wl_list-insert_fake is more better. we care more about what is insert to the list